tests/run-make/libtest-thread-limit/rmake.rs

```diff
@@ -0,0 +1,42 @@
+// libtest used to panic if it hit the thread limit. This often resulted in spurious test failures
+// (thread 'main' panicked at 'called Result::unwrap() on an Err value: Os
+// { code: 11, kind: WouldBlock, message: "Resource temporarily unavailable" }' ...
+// error: test failed, to rerun pass '--lib').
+// Since the fix in #81546, the test should continue to run synchronously
+// if it runs out of threads. Therefore, this test's final execution step
+// should succeed without an error.
+// See https://github.com/rust-lang/rust/pull/81546
+
+//@ only-linux
+// Reason: thread limit modification
+
+use std::ffi::{self, CStr, CString};
+use std::path::PathBuf;
+
+use run_make_support::{libc, run, rustc};
+
+fn main() {
+    rustc().input("test.rs").arg("--test").run();
+
+    let pid = unsafe { libc::fork() };
+    assert!(pid >= 0);
+
+    if pid == 0 {
+        let test = CString::new("test").unwrap();
+        // The argv array should be terminated with a NULL pointer.
+        let argv = [test.as_ptr(), std::ptr::null()];
+        let rlimit = libc::rlimit { rlim_cur: 1, rlim_max: 1 };
+        let ret = unsafe { libc::setrlimit(libc::RLIMIT_NPROC, &rlimit as *const libc::rlimit) };
+        assert!(ret == 0);
+
+        let ret = unsafe { libc::execv(test.as_ptr(), argv.as_ptr()) };
+        assert!(ret == 0);
+    } else {
+        // parent
+
+        let mut child_exit_status: libc::c_int = 0;
+        let ret = unsafe { libc::waitpid(pid, &mut child_exit_status as *mut libc::c_int, 0) };
+        assert!(ret == pid);
+        assert!(child_exit_status == 0);
```

Comment:
Problem [REV1 (1/2)]: (my bad) this check is incorrect w.r.t. the `waitpid` syscall (see [linux.die.net/man/2/waitpid](https://linux.die.net/man/2/waitpid)). We cannot compare this directly, we need to:

1. Check if child process terminated normally with `WIFEXITED(status)` macro in C or `libc::WIFEXITED` function in `libc`.
2. If `libc::WIFEXITED` is true (asserted), then we want to further inspect child exit status with `WEXITSTATUS(status)` macro in C or `libc::WEXITSTATUS`. And only then, we should assert that child exit status is `0` (i.e. success).

Reply:
Good point, forgot about the W_* macros in my code.

Reply:
> Good point, forgot about the W_* macros in my code.

Don't worry, I completely forgor about them too lol.